packages/flutter/test/material/material_test.dart

```diff
@@ -64,6 +64,20 @@ class ElevationColor {
 }
 
 void main() {
+  // Regression test for https://github.com/flutter/flutter/issues/81504
+  testWidgets('MaterialApp.home nullable and update test', (WidgetTester tester) async {
+    // _WidgetsAppState._usesNavigator == true
+    await tester.pumpWidget(const MaterialApp(home: SizedBox.shrink()));
+
+    // _WidgetsAppState._usesNavigator == false
+    await tester.pumpWidget(const MaterialApp()); // Do not crash!
+
+    // _WidgetsAppState._usesNavigator == true
+    await tester.pumpWidget(const MaterialApp(home: SizedBox.shrink())); // Do not crash!
```

Comment:
Actually,this line test the change of the patch that you pointed out.
We should initial the '_navigator' if it equals null during update widget.

Reply:
I also describe it at the description:"Also, fix a crashing when update MaterialApp.home to non-null from null."

Reply:
Got it, thanks for the explanation.